### DISCUSSION TITLE
esp32 keeps rebooting

### DISCUSSION THREAD

**Discussion by 82revo** [renatobo/bonogps#40] [General]
Hello, I am strugeling to get the esp32 started. it keeps reseting. when I enabled the debug mode, I could see more details but still cannot figure out what causes the problem. I tried to reduce the baud rate to 9600 which is the default for M8N.
Any help will be much appreciated. Thank you.
log:

platformio.ini:

[env:esp32dev]
platform = espressif32
framework = arduino
lib_deps = 
	h2zero/NimBLE-Arduino @ ^1.2.0
	yiannisbourkelis/Uptime Library @ ^1.0.0
	evert-arias/EasyButton @ ^2.0.1
	arkhipenko/TaskScheduler @ ^3.2.2
board_build.partitions = min_spiffs.csv
monitor_speed = 115200
lib_ldf_mode = chain+

; basic configuration for a troublshooting/debug build without verbose logging
; OTA is enabled without timeout
board = esp32dev
;monitor_port = /dev/cu.usbserial-0001
;upload_port = /dev/cu.usbserial-0001
board_build.f_flash = 80000000L
build_type = debug
build_flags = 
	-DLOG_LOCAL_LEVEL=ESP_LOG_DEBUG
	-DCORE_DEBUG_LEVEL=4
	-DENABLE_OTA
	-DOTA_AVAILABILITY_SECS=-1
	-DBUILD_ENV_NAME='"Test build - OTA"'

Log:

Executing task in folder bonogps-main: C:\Users\310209838\.platformio\penv\Scripts\platformio.exe device monitor --port COM9 

--- Terminal on COM9 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
[    29][D][esp32-hal-cpu.c:244] setCpuFrequencyMhz(): PLL: 480 / 2 = 240 Mhz, APB: 80000000 Hz
[   247][D][bonogps.cpp:2691] setup(): ESP32 SDK: v4.4.6-dirty        
[   247][D][bonogps.cpp:2692] setup(): Arduino sketch: src/bonogps.cpp
[   248][D][bonogps.cpp:2693] setup(): Compiled on: Apr 15 2024       
[   254][E][Preferences.cpp:483] getString(): nvs_get_str len fail: wifi NOT_FOUND
[   261][D][bonogps.cpp:537] ReadNVMPreferences(): Error reading WiFi, Preference set as WIFI_AP
[   269][E][Preferences.cpp:460] getString(): nvs_get_str len fail: wifikey NOT_FOUND 
[   277][E][Preferences.cpp:460] getString(): nvs_get_str len fail: wifissid NOT_FOUND
[   285][I][bonogps.cpp:2551] bt_spp_stop(): bt stopped
I NimBLEDevice: BLE Host Task Started
I NimBLEDevice: NimBle host synced.      
D NimBLEDevice: >> setPower: 7 (type: 11)
D NimBLEDevice: << setPower
D NimBLEDevice: >> setPower: 7 (type: 9)
D NimBLEDevice: << setPower
D NimBLEDevice: >> setPower: 7 (type: 0)
D NimBLEDevice: << setPower
D NimBLEDevice: >> setPower: 7 (type: 1)
D NimBLEDevice: << setPower
D NimBLEDevice: >> setPower: 7 (type: 2)
D NimBLEDevice: << setPower
D NimBLEDevice: >> setLocalMTU: 185     
D NimBLEDevice: << setLocalMTU
[   585][D][bonogps.cpp:2422] ble_start(): BLE MTU set to: 185
D NimBLEDevice: Setting bonding: 0, mitm: 0, sc: 1
D NimBLEServer: >> createService - 0x1819
D NimBLEServer: << createService
D NimBLEService: >> start(): Starting service: UUID: 0x1819, handle: 0xffff     
D NimBLEService: Adding 1 characteristics for service UUID: 0x1819, handle: 0xffff
D NimBLEService: << start()
D NimBLEServer: >> createService - 0x180a
D NimBLEServer: << createService
D NimBLEService: >> start(): Starting service: UUID: 0x180a, handle: 0xffff
D NimBLEService: Adding 5 characteristics for service UUID: 0x180a, handle: 0xffff
D NimBLEService: << start()
D NimBLEAdvertising: >> Advertising start: customAdvData: 0, customScanResponseData: 0
primary service
           uuid 0x1800
         handle 1
     end_handle 5
characteristic
           uuid 0x2a00
     def_handle 2
     val_handle 3
   min_key_size 0
          flags [READ]
characteristic
           uuid 0x2a01
     def_handle 4
     val_handle 5
   min_key_size 0
          flags [READ]
primary service
           uuid 0x1801
         handle 6
     end_handle 9
characteristic
           uuid 0x2a05
     def_handle 7
     val_handle 8
   min_key_size 0
          flags [INDICATE]
ccc descriptor
           uuid 0x2902
         handle 9
   min_key_size 0
          flags [READ|WRITE]
primary service
           uuid 0x1819
         handle 10
     end_handle 13
characteristic
           uuid 0x2a67
     def_handle 11
     val_handle 12
   min_key_size 0
          flags [READ|NOTIFY]
ccc descriptor
           uuid 0x2902
         handle 13
   min_key_size 0
          flags [READ|WRITE]
primary service
           uuid 0x180a
         handle 14
     end_handle 24
characteristic
           uuid 0x2a29
     def_handle 15
     val_handle 16
   min_key_size 0
          flags [READ]
characteristic
           uuid 0x2a24
     def_handle 17
     val_handle 18
   min_key_size 0
          flags [READ]
characteristic
           uuid 0x2a25
     def_handle 19
     val_handle 20
   min_key_size 0
          flags [READ]
characteristic
           uuid 0x2a26
     def_handle 21
     val_handle 22
   min_key_size 0
          flags [READ]
characteristic
           uuid 0x2a05
     def_handle 23
     val_handle 24
   min_key_size 0
          flags [WRITE_NO_RSP]
D NimBLEAdvertising: << Advertising start
[   799][I][bonogps.cpp:2496] ble_start(): Waiting a client connection to notify...
[   804][I][bonogps.cpp:867] wifi_AP(): Stop Web Portal
[   821][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 0 - WIFI_READY
[   828][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 10 - AP_START
[   831][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 11 - AP_STOP
[  1404][I][bonogps.cpp:882] wifi_AP(): Wait 100 ms for AP_START...
[  1405][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 10 - AP_START
[  1504][D][bonogps.cpp:884] wifi_AP(): Power: 78
[  1504][I][bonogps.cpp:886] wifi_AP(): Set softAPConfig
[  1505][I][bonogps.cpp:892] wifi_AP(): AP IP address: 10.0.0.1
[  1511][I][bonogps.cpp:903] wifi_AP(): mDNS responder started
[  1513][I][bonogps.cpp:907] wifi_AP(): Start Web Portal
[  1534][D][bonogps.cpp:911] wifi_AP(): Start blinking slowly 1sec frequency
[  1534][D][bonogps.cpp:2589] gps_initialize_settings(): Send a ping to start the GPS if it was powersaved
[  2053][D][bonogps.cpp:2598] gps_initialize_settings(): Start UART connection on RX pin 16 TX pin 17 and autobaudrate
[  2161][D][bonogps.cpp:2602] gps_initialize_settings(): Connected with autobaudrate at 9600 on RX pin 16 TX pin 17 
[  2361][E][HardwareSerial.cpp:614] setRxBufferSize(): RX Buffer can't be resized when Serial is already running.

[  2413][D][bonogps.cpp:2766] setup(): Total heap: 243648
[  2413][D][bonogps.cpp:2767] setup(): Free heap: 122400
[  2413][D][bonogps.cpp:2768] setup(): Total PSRAM: 0
[  2417][D][bonogps.cpp:2769] setup(): Free PSRAM: 0
[  2421][W][bonogps.cpp:2784] loop(): New max buffer: 251
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d3c9e  PS      : 0x00060d30  A0      : 0x800db8ed  A1      : 0x3ffd0ca0
A2      : 0x00000001  A3      : 0x3ffc4680  A4      : 0x00000000  A5      : 0x00000ae0
A6      : 0x3f4032e4  A7      : 0x000000fb  A8      : 0x800f22ca  A9      : 0x3ffd0c60
A10     : 0x0000097b  A11     : 0x00000000  A12     : 0x00000000  A13     : 0x00000000
A14     : 0x3f40327c  A15     : 0x00000000  SAR     : 0x00000000  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x000009ab  LBEG    : 0x40085059  LEND    : 0x40085061  LCOUNT  : 0x00000027  

Backtrace: 0x400d3c9b:0x3ffd0ca0 0x400db8ea:0x3ffd0cc0 0x40102809:0x3ffd0d00

ELF file SHA256: fc89c4bd626f9521

Rebooting...
ets Jul 29 2019 12:21:46

**Comment by renatobo** [comment]
Hello, thanks for taking the time to log all the information.
9600 is not enough to support all the messages from the GPS, you could test 38400 at 1Hz, but I recommend the standard 115200.

**Comment by renatobo** [comment]
I recently had the same issue due to evert-arias/EasyButton version 2.0.3 (and wasting a Saturday because of it), it looks like it might be the same issue so I'll mark this close for now.
Version 1.4 just released blocks EasyButton to version 2.0.1 which works, I'll look into a substitute library as this is no longer maintained.
